### PR TITLE
Readable description for "InsufficientFundsForRent"

### LINKED ISSUE #250: Readable description for "InsufficientFundsForRent"
Currently, the error message for "insufficient funds for rent" is parsed as "Unknown Error." Since this is a common error, it'd be nice to make the error more readable.

#### Original ([reference tx](https://explorer.solana.com/tx/3CbFNhwwCEo4dajUcaHf7Eee6A5HHcWUwuPQxZGHGycifjmZUbESULE37oWUw6YUUBuQYrsKzQTreiMDMaMsc5YU))
Screenshot 2023-05-31 at 4 52 28 PM

#### Improved
Screenshot 2023-05-31 at 4 47 30 PM

cc @ngundotra

### PR DESCRIPTION
**Pull request by jdubpark**
Fixes #250

### CODE REVIEW

File: app/tx/[signature]/page-client.tsx [hunk 1]
```diff
@@ -1,34 +1,33 @@
...
+import {ProgramLogSection} from '@components/transaction/ProgramLogSection';
+import {TokenBalancesCard} from '@components/transaction/TokenBalancesCard';
+import {FetchStatus} from '@providers/cache';
+import {useCluster} from '@providers/cluster';
+import {useFetchTransactionStatus, useTransactionDetails, useTransactionStatus} from '@providers/transactions';
+import {useFetchTransactionDetails} from '@providers/transactions/parsed';
+import {SystemInstruction, SystemProgram, TransactionSignature} from '@solana/web3.js';
+import {Cluster, ClusterStatus} from '@utils/cluster';
+import {displayTimestamp} from '@utils/date';
+import {SignatureProps} from '@utils/index';
```
Comment (ngundotra) [inline app/tx/[signature]/page-client.tsx]: Can you run `yarn fmt` on your code & fix these no-op changes?

These changes do nothing to affect code, but affect readability of the PR

File: app/utils/tx.ts [hunk 2]
```diff
@@ -558,3 +560,30 @@ export function intoParsedTransaction(tx: Transaction): ParsedTransaction {
         signatures: tx.signatures.map(value => bs58.encode(value.signature as any)),
     };
 }
+
+export function intoTransactionErrorReason(info: TransactionStatusInfo, tx?: ParsedTransaction | Transaction): { errorReason: string, errorLink?: string } {
```
Comment (ngundotra) [inline app/utils/tx.ts]: Can you rename this to `getTransactionErrorReason` & move this back to the page it came from?

It's not sufficiently general to be moved into `utils`

**Review by ngundotra — CHANGES_REQUESTED** [review]

File: app/tx/[signature]/page-client.tsx (hunk 1 above)
Comment (jdubpark) [reply app/tx/[signature]/page-client.tsx]: Sorry, fixed the format. I don't know why my IDE is doing that (fixed by using vscode now)

File: app/tx/[signature]/page-client.tsx [hunk 3]
```diff
@@ -48,6 +53,34 @@ type Props = Readonly<{
...
+        if (tx) {
+            const address = tx.message.accountKeys[accountIndex].pubkey;
+            return { errorLink: `/address/${address}`, errorReason: `Insufficient Funds For Rent: ${address}` };
+        }
+        return { errorReason: `Insufficient Funds For Rent: Account #${accountIndex + 1}` };
+    } catch (e) {
+        // ignore (return below)
+    }
+
+    return { errorReason: `Unknown Error: "${JSON.stringify(info.result.err)}"` }; // catch-all
```
Comment (ngundotra) [inline app/tx/[signature]/page-client.tsx]: Nit: Can you remove the comments from this function?

File: app/utils/tx.ts [hunk 4]
```diff
@@ -448,7 +448,7 @@ export const PROGRAM_INFO_BY_ID: { [address: string]: ProgramInfo } = {
     },
 };
 
-export type LoaderName = typeof LOADER_IDS[keyof typeof LOADER_IDS];
+export type LoaderName = (typeof LOADER_IDS)[keyof typeof LOADER_IDS];
```
Comment (ngundotra) [inline app/utils/tx.ts:451]: Nit: Can you remove the parantheses here?

File: app/tx/[signature]/page-client.tsx [hunk 5]
```diff
@@ -229,7 +263,13 @@ function StatusCard({ signature, autoRefresh }: SignatureProps & AutoRefreshProp
...
                         <td className="text-lg-end">
                             <h3 className="mb-0">
-                                <span className={`badge bg-${statusClass}-soft`}>{errorReason}</span>
+                                {errorLink ? (
+                                    <Link href={errorLink}>
+                                        <span className={`badge bg-${statusClass}-soft`}>{errorReason}</span>
+                                    </Link>
+                                ) : (
+                                    <span className={`badge bg-${statusClass}-soft`}>{errorReason}</span>
+                                )}
```
Comment (ngundotra) [inline app/tx/[signature]/page-client.tsx]: This actually works surprisingly well. No UI indicator mentioning it's a link, but still great.

**Review by ngundotra — CHANGES_REQUESTED** [review]
Minor nits related to code hygiene, but otherwise looks great & think it's ready to ship!

cc @steveluscher

File: app/utils/tx.ts (hunk 4 above)
Comment (steveluscher) [reply app/utils/tx.ts:451]: I think this is actually a new [Prettier feature](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEcAeAHCAnGACGATwzjwBkIBDAEzmwDlKBbUgXgOLggDNyB5AIIARAKIAlAPoBJIQGUA2gGs4hHhxJqyg0ZJmyAugG4QAGhAQMMAJbQAzslCVs2CAHcACk4T2UlADaulIT2ZgBG2JRgyjCyzHBkVlBwyNz+tnBhEVFwMRiRiQDmyDDYAK4ZqEyhcNS01GSUUAWllAVwAGI4TJQw1k3IIJSlMBCmIAAWMEx+AOrjVvC2eWBwst4LVgBuC4QDYLYhIInpuO4RBd0paRUAVrZosoV+cACKpRDwV37pZnnYJwMiCRbGBsFZLGMMGDYDMrNQYONkAAOAAMvxc6RmEQwAyhcBOm2SZgAju94GcLD5BrYALRJGo1MbYOCkqzMs6tS5IVLfCrpJhWYplPlPOACXpg0LDODuOgJJJfH4gWyit4fZLc65mGCUUKw+GIpAAJm1ESsfkKAGEIEwuahbABWMaldIAFV1Ph5Ss25SkUFosFkoPBMAE-tkRGeirgAF8Y0A). We should upgrade Prettier to the latest and make these changes.

**Comment by jdubpark** [comment]
Removed comments and removed the try-catch in `getTransactionErrorReason` for (future) other error catches.

Also, added `err.errorLink` validation to only link errors when a link is not undefined. Previously it led to errors linking to `/undefined`.

**Review by jdubpark — COMMENTED** [review]
Fixed accordingly

**Pull request merged**